### PR TITLE
Skip demos in docs build that do not produce figures

### LINKED ISSUE #62: Don't run all demos when building docs
When building docs we need to run demos only to save the figures in them, right? If so, we could significantly cut down the time to build docs by not running demos without figures in them.

### PR DESCRIPTION
**Pull request by ddundo**
Closes #62 

Cuts time by about 12 min :)

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -29,16 +29,28 @@ run_animate_demos:
 	cd . \
 
 run_goalie_demos:
+	@SKIP_DEMOS="time_partition.py gray_scott.py gray_scott_split.py"; \
```
Comment (jwallwork23) [inline docs/Makefile:32]: What does the `@` mean here? I haven't seen this notation before and as far as I can tell it is redundant.

File: docs/Makefile [hunk 2]
```diff
@@ -29,16 +29,28 @@ run_animate_demos:
...
-		cd  ${GOALIE}/demos/ && \
-		${PYTHON} $$FILE ; \
+		BASENAME=$$(basename $$FILE); \
+		if ! echo "$$SKIP_DEMOS" | grep -wq $$BASENAME; then \
+			cd ${GOALIE}/demos/ && \
+			${PYTHON} $$FILE; \
+		else \
+			echo "Skipping Goalie demo $$BASENAME"; \
+		fi; \
 	done; \
```
Comment (jwallwork23) [inline docs/Makefile:41]: Not specific to your change, but how about we simplify things slightly by only changing directory once?
```sh
cd ${GOALIE}/demos; \
for FILE in *.py; do \
  ... \
done
```
and similarly for the Animate and Movement demos.

**Review by jwallwork23 — CHANGES_REQUESTED** [review]
Nice, always good to cut down the test suite time.

I have one question and a suggested simplification.

File: docs/Makefile (hunk 1 above)
Comment (ddundo) [reply docs/Makefile]: It just means that whatever is after `@` doesn't get echoed to console. Minor :) I removed the `@` in https://github.com/mesh-adaptation/docs/pull/92/commits/7906cf4abfaafd8a205170598c4f63a44450aefb

Comment (jwallwork23) [reply docs/Makefile]: Ah yes I did know that. Thanks!

File: docs/Makefile (hunk 2 above)
Comment (ddundo) [reply docs/Makefile]: Nice! Done that now in the last 2 commits :)

**Review by jwallwork23 — APPROVED** [review]
Looks good, thanks!

**Pull request merged**